lib/neographics/src/common.c

```diff
@@ -31,7 +31,10 @@ void n_graphics_set_pixel(n_GContext * ctx, n_GPoint p, n_GColor color) {
         &ctx->fbuf[p.y * __SCREEN_FRAMEBUFFER_ROW_BYTE_AMOUNT + p.x / 8],
         p.x % 8, (color.argb & 0b111111));
 #else
-    ctx->fbuf[p.y * __SCREEN_FRAMEBUFFER_ROW_BYTE_AMOUNT + p.x] = color.argb;
+    if (p.y * __SCREEN_FRAMEBUFFER_ROW_BYTE_AMOUNT + p.x >= 0)
```

Comment:
hmm ... any chance we could take this off the hot path?  (do bounding in the outer loop?)